orchestration/transfer_controller.py

```diff
@@ -0,0 +1,223 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from dotenv import load_dotenv
+from enum import Enum
+import logging
+import os
+import time
+from typing import Generic, Protocol, TypeVar
+
+import globus_sdk
+
+from orchestration.flows.bl832.config import Config832
+from orchestration.globus.transfer import GlobusEndpoint, start_transfer
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+load_dotenv()
+
+
+Endpoint = TypeVar("Endpoint", bound=GlobusEndpoint)
+
+
+class BaseEndpoint(Protocol):
```

Comment:
Protocols and ABCs do similar jobs in different ways. In part it's a matter of style which you choose. I prefer ABCs because I like inheritence and runtime enforcement. But I think we should be consistent in this PR and only use one or the other. Can you change this to ABCs?

FWIW, it doesn't make much sense to mark the methods in this class as `abstractmethod` for a Protocol.